src/api.js

```diff
@@ -844,6 +962,63 @@ Module["onRuntimeInitialized"] = function onRuntimeInitialized() {
         return stmt;
     };
 
+    /** Iterate over multiple SQL statements in a SQL string
+     @param {string} sql a string of SQL
+     @return {StatementIterator} the resulting statement iterator
+     */
+    Database.prototype["iterateStatements"] = function iterateStatements(sql) {
+        return new StatementIterator(sql, this);
+    };
+
+    /* Internal methods to implement statement iteration */
+
+    Database.prototype.advanceIterator = function advanceIterator(iter) {
+        if (iter.sqlPtr === null) {
+            throw "Invalid iterator";
+        }
+        if (iter.activeStatement !== null) {
+            iter.activeStatement["free"]();
+            iter.activeStatement = null;
+        }
+        if (!this.db) {
+            this.finalizeIterator(iter);
+            throw "Database closed";
+        }
+        var stack = stackSave();
+        var pzTail = stackAlloc(4);
+        setValue(apiTemp, 0, "i32");
+        setValue(pzTail, 0, "i32");
+        var returnCode = sqlite3_prepare_v2_sqlptr(
+            this.db,
+            iter.nextSqlPtr,
+            -1,
+            apiTemp,
+            pzTail
+        );
+        if (returnCode === SQLITE_OK) {
+            iter.nextSqlPtr = getValue(pzTail, "i32");
+        }
+        stackRestore(stack);
+        if (returnCode === SQLITE_OK) {
+            var pStmt = getValue(apiTemp, "i32");
+            if (pStmt === NULL) {
+                this.finalizeIterator(iter);
+                return { done: true };
+            }
+            iter.activeStatement = new Statement(pStmt, this);
+            this.statements[pStmt] = iter.activeStatement;
+            return { value: iter.activeStatement, done: false };
+        }
+        iter.nextSqlString = UTF8ToString(iter.nextSqlPtr);
```

Comment:
What is the point of setting the private nextSqlString field here ?

Reply:
So that I can free the memory pointed to by nextSqlPtr and still have access to the string.